vimv

```diff
@@ -21,18 +21,27 @@ done
 ${EDITOR:-vi} "${FILENAMES_FILE}"
 
 IFS=$'\r\n' GLOBIGNORE='*' command eval  'dest=($(cat "${FILENAMES_FILE}"))'
-
-count=0
+renamed=0
+deleted=0
 for ((i=0;i<${#src[@]};++i)); do
     if [ "${src[i]}" != "${dest[i]}" ]; then
         mkdir -p "`dirname "${dest[i]}"`"
         if git ls-files --error-unmatch "${src[i]}" > /dev/null 2>&1; then
             git mv "${src[i]}" "${dest[i]}"
+            ((renamed++))
+        elif [ "del" = "${dest[i]}" ]; then # in vim replace the file name with 'del' to delete the file/dir
+            if [ ! -z $VIMV_USE_RMTRASH ]; then # export VIMV_USE_RMTRASH=1 to use rmtrash
+              rmtrash -rf "${src[i]}"
+            else
+              rm -rf "${src[i]}" 
+            fi
+            ((deleted++))
```

Comment:
This will fail inside git repositories, `git rm` should be used to make the
interface uniform inside and outside git repos

Reply:
thanks for reviewing my pull.
can i use `if git ls-files --error-unmatch "${src[i]}" > /dev/null 2>&1; then` to check if the file is tracked by git ? somethings like this : 
```
        if git ls-files --error-unmatch "${src[i]}" > /dev/null 2>&1; then
            git mv "${src[i]}" "${dest[i]}"
            ((renamed++))
        elif [ "del" = "${dest[i]}" ]; then # in vim replace the file name with 'del' to delete the file/dir
            if git ls-files --error-unmatch "${src[i]}" > /dev/null 2>&1; then
              git rm "${src[i]}" # correct if you have a better command to rm the file
            elif [ ! -z $VIMV_USE_RMTRASH ]; then # export VIMV_USE_RMTRASH=1 to use rmtrash
              rmtrash -rf "${src[i]}"
            else
              rm -rf "${src[i]}" 
            fi
            ((deleted++))
```

Reply:
No problem, the `del` check must happen before we check the current path is a
repository, otherwise `git mv` it will try to rename the file as `del` before the
`elif [ del = "${dest[i]}" ]` check

Reply:
understood, I will do a commit tomorrow, thanks